apps/sodo-search/src/search-index.js

```diff
@@ -133,4 +137,79 @@ export default class SearchIndex {
             tags: this.#normalizeSearchResult(tags)
         };
     }
+
+    checkLocaleCjk(locale) {
+        if (!locale) {
+            return false;
+        }
+        // Chinese
+        if (locale === 'zh' || locale === 'zh-Hans' || locale === 'zh-Hant') {
+            return true;
+        }
+        // Japanese
+        if (locale === 'ja') {
+            return true;
+        }
+        // Korean
+        if (locale === 'ko') {
+            return true;
+        }
+        return false;
+    }
+
+    getCjkFlexsearchDocumentPatchOptions(locale) {
+        const isLocaleCjk = (() => {
+            if (!locale) {
+                return false;
+            }
+            // Chinese
+            if (locale === 'zh' || locale === 'zh-Hans' || locale === 'zh-Hant') {
+                return true;
+            }
+            // Japanese
+            if (locale === 'ja') {
+                return true;
+            }
+            // Korean
+            if (locale === 'ko') {
+                return true;
+            }
+            return false;
+        })();
+
+        if (!isLocaleCjk) {
+            return {};
+        }
+
+        return {
+            encode: (str) => {
+                // CJK character regex
+                const regexCjk = new RegExp(`([${
```

Comment:
Since this is a static value, can you define it at the top? Without a perf profile, it's hard to say, but I have a feeling that moving it out will give you some free optimization.

Reply:
Okay, I think so too.